File: games/smite/item.py
import dataclasses


@dataclasses.dataclass
class Item:
    name: str


@dataclasses.dataclass
class Starter(Item):
    pass


@dataclasses.dataclass
class Build:
    item1: Starter
    item2: Item
    item3: Item
    item4: Item
    item5: Item
    item6: Item

File: src/smite/item.cpp
#include "smite/item.h"

namespace twoptimizer::smite {

Item::Item(const std::string& name, const Stats& stats, bool isStarter) 
    : name_(name), stats_(stats), is_starter_(isStarter) {
}

int Build::countItems() const {
    int count = 0;
    if (item1) count++;
    if (item2) count++;
    if (item3) count++;
    if (item4) count++;
    if (item5) count++;
    if (item6) count++;
    return count;
}

} // namespace twoptimizer::smite
